cordapp-cpk/src/main/kotlin/net/corda/plugins/cpb/CpbTask.kt

```diff
@@ -38,6 +53,11 @@ open class CpbTask : Jar() {
         isPreserveFileTimestamps = false
         isReproducibleFileOrder = true
         isZip64 = true
+
+        manifest { m ->
+            m.attributes[CPB_NAME_ATTRIBUTE] = cpbName
+            m.attributes[CPB_VERSION_ATTRIBUTE] = cpbVersion
+        }
```

Comment:
I have fought mighty battles with Gradle over something that looked _very_ similar to this - see the comment in `PackagingTask`. My use-case _may_ have been more complicated though, e.g. I was trying to extract manifest tags from a jar built by another task, and then Gradle complained that it couldn't serialise my lazy properties into the configuration cache... :nauseated_face:.

Reply:
I was also very concerned that Gradle might end up writing something like `ord.gradle.api.provider.Property@60c1663c` in my manifest but it *seems* to handle it correctly though.. Do you think there is any other testing that would need to be performed?

Reply:
the other thing is that I speculate (but I'm not in any way sure) that the `manifest` configuration closure is executed at some point in the configuration phase, so I couldn't use `cpbName.get()` as the property wasn't guaranteed to have been populated

Reply:
> the other thing is that I speculate (but I'm not in any way sure) that the `manifest` configuration closure is executed at some point in the configuration phase

More likely, Gradle is just "helpfully" executing _after_ the manifest has already been written :facepalm:.

Reply:
well, if that was the case I wouldn't see my attributes in the manifest, but I have at least a single test case that performs correctly

Reply:
You are adding these attributes in the task's constructor, which is "early enough". However, the manifest _should_ also be configurable via a lazy action executed via (say) `doFirst {}` etc.